app/controllers/listings_controller.rb

```diff
@@ -179,8 +183,10 @@ def create
     end
 
     params[:listing] = normalize_price_param(params[:listing]);
+    shape = get_shape(Maybe(params)[:listing][:transaction_type_id].to_i.or_else(nil))
+    unit_type = Maybe(shape[:units].first)[:type].or_else(nil)
```

Comment:
This is only temporary. Currently we save the unit to `transaction_types` table and to the `units` table. Because `transaction_types` can have only one unit we save only the first one to the units table.